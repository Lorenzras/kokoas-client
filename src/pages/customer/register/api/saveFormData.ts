import { APPIDS, KintoneRecord } from '../../../../api/kintone';
import { CustomerForm } from '../form';
import { formToKintConst } from '../helper';
import { saveCustomers } from './saveCustomers';
import { updateProjDetails } from './updateProjDetails';

export const saveFormData = async (formData: CustomerForm) => {
  try {
    const savedCustomers = await saveCustomers(formData);
    const transformedForm = formToKintConst(formData, savedCustomers);

    console.log(savedCustomers, 'savedCustomers');

    let saveResult: { id?: string, revision: string }  = Object.create(null);

    /* Create record */
    if (!formData.id) {
      saveResult = await KintoneRecord.addRecord({
        app: APPIDS.custGroup,
        record: transformedForm,
      });
      return saveResult;
    } else {
      saveResult = await KintoneRecord.updateRecord({
        app: APPIDS.custGroup,
        id: formData.id,
        record: transformedForm,
      })
        .then((resp)=> ({ id: formData.id, revision: resp.revision }));
    }

    /* Copy information to project details */
    await updateProjDetails(formData);

    return saveResult;
  } catch (err) {
    throw new Error(err.message);
  }
};