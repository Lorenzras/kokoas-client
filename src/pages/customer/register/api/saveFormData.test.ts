import { saveFormData } from './saveFormData';
import { testData } from './test/testData';

test('Save', async ()=>{
  expect(await saveFormData(testData)).toMatchSnapshot();
});