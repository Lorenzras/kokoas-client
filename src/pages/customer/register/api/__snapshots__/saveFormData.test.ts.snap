// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save 1`] = `
Object {
  "id": "50",
  "revision": "1",
}
`;
